### PR TITLE
Remove hard-coded DLLs from Windows installer build

### PR DESCRIPTION
**Pull request by swt2c**
The set of bundled Windows DLLs had been hard-coded in the NSIS script
and thus it was difficult to maintain.  Instead, use mingw-ldd to
automatically generate the list of required DLLs and copy them to a temp
directory before creating the installer.

### CODE REVIEW

File: .github/workflows/ci-windows.yml [hunk 1]
```diff
@@ -24,3 +24,23 @@ jobs:
...
+    runs-on: windows-latest
+    steps:
+    - uses: actions/download-artifact@v2
+      id: download
+      with:
+        name: windows-installer
+    - name: Install and test
+      run: |
+        ${{steps.download.outputs.download-path}}\concordance-installer.exe /S
+        "C:\Program Files (x86)\Concordance\concordance.exe --version"
```
Comment (jaymzh) [inline .github/workflows/ci-windows.yml:46]: <3

File: concordance/win/make_installer.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/python3
```
Comment (jaymzh) [inline concordance/win/make_installer.py:1]: This seems like one of the few times a shell script might be simpler, but I have no objections to this.

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Thanks for all the CI work man, it's awesome. Been meaning to do it forever.

File: concordance/win/make_installer.py (hunk 2 above)
Comment (swt2c) [reply concordance/win/make_installer.py:1]: Yeah, I was on the fence about it.  I like Python's `tempfile.TemporaryDirectory` with a context manager though, so I don't have to worry about cleaning it up.

File: .github/workflows/ci-windows.yml (hunk 1 above)
Comment (swt2c) [reply .github/workflows/ci-windows.yml:46]: I'm not convinced that this actually worked though as I don't see anything in the output.  Guess I should have marked this [WIP] as I was still working on it.  :-)

Comment (jaymzh) [reply .github/workflows/ci-windows.yml:46]: You know you can mark a PR as "draft" in GH as well, right?

Comment (jaymzh) [reply .github/workflows/ci-windows.yml:46]: image

Comment (swt2c) [reply .github/workflows/ci-windows.yml:46]: Yeah, I knew that but I couldn't figure out how to do it.  Figured it out now.

**Pull request merged**